src/app/body/all-products/product-card/product-card.component.html

```diff
@@ -38,10 +38,10 @@
       <div class="product-card_details">
         <a routerLink="/products/{{product.id}}">Product Detail</a>
       </div>
-      <div class="product-card_wishlist">
-        <button>
-          <img src="assets/heart.svg" alt="">
-          Add to wish list</button>
+      <div class="product-card_wishlist" *ngIf="authService.user.value.id && !wishlist">
```

Comment:
You shouldn't have access to the service data like this. Your service should always have a private variable

Reply:
fixed